### PR TITLE
Move `seed-monitoring` and `shoot-core` charts and monitoring-related logic in `Shoot` controller into `monitoring` component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
Similar to https://github.com/gardener/gardener/pull/8237, this PR moves the `seed-monitoring` and `shoot-core` charts and monitoring-related logic in `Seed` controller into `monitoring` component for better dev-productivity (incl. usage of go embed).

On the way, the other following improvements were made:
- dropped usage of `operation/common` package (only loki->vali migration logic remains which will be cleaned up soon)
- embedded remainings of `shoot-core` chart (PSPs only, will be dropped once minimum Kubernetes version will be raised to 1.25)
- eliminated `utils-templates` chart (which was sym-linked a lot)

Part of https://github.com/gardener/gardener/issues/6016

**Special notes for your reviewer**:
This PR explicitly just moves code and does not introduce tests or usage of `ManagedResource` since all this should be replaced in the scope of GEP-19.

Depends on the following PR, hence this PR remains in draft state until it is merged.
- [x] https://github.com/gardener/gardener/pull/8237

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/component/monitoring/component.go [hunk 1]
```diff
@@ -55,6 +55,15 @@ var (
 	chartPathCore = filepath.Join("charts", "seed-monitoring", "charts", "core")
 )
 
+// Interface contains functions for a mointoring deployer.
```
Comment (shafeeqes) [inline pkg/component/monitoring/component.go]:
```suggestion
// Interface contains functions for a monitoring deployer.
```

**Review by shafeeqes — COMMENTED** [review]
Awesome PR ❤️ 
Only one minor nit.

File: pkg/component/monitoring/component.go [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+	StorageCapacityAlertmanager string
+	// TargetName is the name of the target cluster.
+	TargetName string
+	// TargetProviderType is the provider type of the target cluster.
+	TargetProviderType string
+	// WildcardCertName is name of wildcard tls certificate which is issued for the seed's ingress domain.
+	WildcardCertName *string
+}
+
+// New creates a new instance of DeployWaiter for the monitoring components.
```
Comment (acumino) [inline pkg/component/monitoring/component.go]:
```suggestion
// New creates a new instance of Deployer for the monitoring components.
```

File: pkg/operation/operation.go [hunk 3]
```diff
@@ -434,27 +434,27 @@ func (o *Operation) WantsPlutono() bool {
 
 // ComputeKubeAPIServerHost computes the host with a TLS certificate from a trusted origin for KubeAPIServer.
 func (o *Operation) ComputeKubeAPIServerHost() string {
-	return o.ComputeIngressHost(common.KubeAPIServerPrefix)
+	return o.ComputeIngressHost("api")
```
Comment (acumino) [inline pkg/operation/operation.go:437]: Isn't it better to keep the constants for `api` and similar for others?

File: pkg/component/monitoring/component.go [hunk 4]
```diff
@@ -53,6 +55,8 @@ var (
 	//go:embed charts/seed-monitoring/charts/core
 	chartCore     embed.FS
 	chartPathCore = filepath.Join("charts", "seed-monitoring", "charts", "core")
+
+	managedResourceNamePrometheus = "shoot-core-prometheus"
```
Comment (acumino) [inline pkg/component/monitoring/component.go:59]: Do we not need to ignore the prometheus resources in `shoot-core` managed resource? How will migration look like here? `shoot-core` MR will try to delete the resource and `shoot-core-prometheus` MR will try to apply the same resource, is this not an issue?

File: pkg/operation/operation.go (hunk 3 above)
Comment (rfranzke) [reply pkg/operation/operation.go:437]: They were only used here, so there is no need to keep them with just a single usage.

File: pkg/component/monitoring/component.go (hunk 4 above)
Comment (rfranzke) [reply pkg/component/monitoring/component.go:59]: No migration needed here - I renamed the resources, so the old RBAC will disappear (due to `shoot-core` `ManagedResource` reconciliation), and the new RBAC will appear in parallel.

Comment (acumino) [reply pkg/component/monitoring/component.go:59]: Ok, I didn't notice the resource name change.

**Review by acumino — COMMENTED** [review]
/lgtm
/approve

**Comment by rfranzke** [comment]
Thanks @acumino and @shafeeqes, appreciated!

**Pull request merged**